### PR TITLE
add PHP 8.1 support

### PR DESCRIPTION
**Pull request by seyon**
Q | A
-- | --
New Feature | yes

<h3 dir="auto" style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 1.25em; font-weight: 600; line-height: 1.25; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Description</h3><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px !important; color: rgb(201, 209, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">add PHP 8.1 support.</p>

### CODE REVIEW

**Comment by Ocramius** [comment]
The upgrade of `vimeo/psalm` (and subsequent downgrade in 69d20fd) caused some failures:

```
Error: Argument 1 of Laminas\ConfigAggregatorParameters\LazyParameterPostProcessor::__construct expects callable():array<string, mixed>, parent type pure-Closure():array<array-key, mixed> provided (see https://psalm.dev/194)
```

File: composer.json [hunk 1]
```diff
@@ -17,10 +17,11 @@
     "extra": {
     },
     "require": {
-        "php": "^7.3 || ~8.0.0",
+        "php": "^7.4 || ~8.0.0 || ~8.1.0",
         "laminas/laminas-config-aggregator": "^1.1",
         "laminas/laminas-stdlib": "^3.1",
-        "symfony/dependency-injection": "^3.4.36 || ^4.4.2 || ^5.0"
+        "psr/container": "1.1.1",
```
Comment (boesing) [inline composer.json]: Please remove this dependency as it should not be needed

File: composer.json [hunk 2]
```diff
@@ -17,10 +17,11 @@
...
+        "symfony/dependency-injection": "^3.4.36 || ^4.4.2 || ^5.0 || ^6.0"
```
Comment (boesing) [inline composer.json]: please do not add symfony 6 if its not required to have support for PHP 8.1

Do we have any issues if we do not make this bump?

**Review by boesing — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (boesing) [reply composer.json]: Just to clarify. We do not actually verify if all these symfony versions are working in here as we do not have a matrix tto enforce this.
So pretty sure that only 3.4.36 & 5.4 (and something in between, but pretty sure not 4.x) are tested at all and if we now bump to 6.0+, we won't even test 5.4.

Comment (Ocramius) [reply composer.json]: IMO worth bumping to `^5.0 || ^6.0` anyway - if `^6.0` is not installed at all (even in CI), then this diff is mostly fluff :\

Comment (seyon) [reply composer.json]: The whole thing is compatible with 7.2+ and we have a project where we need "symfony/dependency-injection" with 6.0. Therefore, I had increased this directly. Would that cause a problem?

Comment (boesing) [reply composer.json]: I'd be fine with changing to `^5.4 || ^6.0` since 5.4 has LTS and thus we can drop anything below that.

**Review by boesing — APPROVED** [review]
LGTM.

**Pull request merged**